File: NBE_5_7_2_02TEAM/src/main/java/io/twogether/nbe_5_7_2_02team/post/dao/LikesRepository.java
package io.twogether.nbe_5_7_2_02team.post.dao;

import io.twogether.nbe_5_7_2_02team.member.domain.Member;
import io.twogether.nbe_5_7_2_02team.post.domain.Likes;
import io.twogether.nbe_5_7_2_02team.post.domain.Post;

import org.springframework.data.jpa.repository.JpaRepository;

import java.util.Optional;

public interface LikesRepository extends JpaRepository<Likes, Long> {

    Optional<Likes> findByPostAndMember(Post post, Member member);

    void deleteByPost(Post post);
}

File: NBE_5_7_2_02TEAM/src/main/java/io/twogether/nbe_5_7_2_02team/post/dao/LikesRepository.kt
package io.twogether.nbe_5_7_2_02team.post.dao

import io.twogether.nbe_5_7_2_02team.member.domain.Member
import io.twogether.nbe_5_7_2_02team.post.domain.Likes
import io.twogether.nbe_5_7_2_02team.post.domain.Post
import org.springframework.data.jpa.repository.JpaRepository

interface LikesRepository : JpaRepository<Likes, Long> {
    fun existsByPostAndMember(
        post: Post,
        member: Member,
    ): Boolean

    fun findByPostAndMember(
        post: Post,
        member: Member,
    ): Likes?

    fun deleteByPost(post: Post)
}
